frontend/src/layouts/Dapp.js

```diff
@@ -1,36 +1,79 @@
 import React, {useState, useEffect} from "react";
-import {authenticate} from "ceramic/index.js";
 import Sidebar from "components/Sidebar.js";
 import WalletInfo from "components/WalletInfo.js";
 
 import {useLocation} from "react-router-dom";
 
 import {DappContextProvider} from "context/dappContext";
 
+import {getServerDID} from "api/index";
+
+import {readOnlyClient, authenticatedClient} from "identity";
+import {connect} from "ethereum";
+
 const Dapp = ({children}) => {
-  /* Ceramic Code */
+  /* --- Ceramic Code --- */
+
+  // Read Only Ceramic + IDX
+  const [readOnlyClients, setReadOnlyClients] = useState({});
+  async function getReadOnlyClients() {
+    const {ceramic, idx} = await readOnlyClient();
+    setReadOnlyClients((prevState) => {
+      return {...prevState, ...{ceramic, idx}};
+    });
+  }
+
+  // getReadOnly instance on mounted
+  useEffect(() => {
+    getReadOnlyClients();
```

Comment:
From my understanding, useEffect will be call every time a component mount AND update, no ?

If it's the case, then we are calling `readOnlyClient()` again and a new instance of IDX will be creating (_from src/identity/index.js - line 99_).
Does this make sense instead : 
```javascript
useEffect(() => {
    // this check if object is empty. getReadOnlyClients should only be called on mounted
    if (Object.keys(readOnlyClients).length === 0) {
      getReadOnlyClients();
    }
}, []);
```